tools/kubernetes/migration.py

```diff
@@ -0,0 +1,196 @@
+        
+        response_data = response.json()
+        versions.extend(response_data.get("pipeline_versions", [])) 
+        page_token = response_data.get("next_page_token")
+        if not page_token:
+            break
+
+    return versions
+
+# Creates a Kubernetes-safe name for a pipeline version by combining the pipeline name and version ID.
+# If the name is too long, it shortens it and adds index at the end to keep it unique.
+def get_version_name(pipeline_name, version_display_name, index, add_prefix):
+    if add_prefix:
+       original_version_name = f"{pipeline_name}-{version_display_name}"
+    else:
+        original_version_name = f"{version_display_name}"    
+    
+    # Clean and convert the pipeline name to be k8s-compatible
+    formatted_name = utils.maybe_rename_for_k8s(original_version_name)
+
+    if len(formatted_name) > MAX_NAME_LENGTH:
+        suffix = f"-{index}"
+        max_name_len = MAX_NAME_LENGTH - len(suffix)
+        trimmed_name = original_version_name[:max_name_len]
+        formatted_name = utils.maybe_rename_for_k8s(trimmed_name + suffix)
+
+    return formatted_name
+
+# Convert a pipeline and its versions into k8s format
+def convert_to_k8s_format(pipeline, pipeline_versions, add_prefix, namespace):
+    k8s_objects = []
+
+    # Use namespace from pipeline object if available; otherwise fallback to default
+    namespace = pipeline.get('namespace') or namespace
+    original_id = pipeline.get('pipeline_id')
+    display_name = pipeline.get('display_name')
+
+    # Clean and convert the pipeline name to be k8s-compatible
+    pipeline_name = utils.maybe_rename_for_k8s(pipeline.get('display_name', 'unnamed'))
+    versions = pipeline_versions
+
+    # Create the Pipeline object
+    pipeline_obj = {
+        "apiVersion": K8S_PIPELINE_API_VERSION,
+        "kind": "Pipeline",
+        "metadata": {
+            "name": pipeline_name,
+            "namespace": namespace,
+            "annotations": {
+                "pipelines.kubeflow.org/original-id": original_id,
+                "pipelines.kubeflow.org/display-name": display_name
+            }
+        }
+    }
+    k8s_objects.append(pipeline_obj)
+
+     # Create a PipelineVersion object for each version
+    for i, version in enumerate(versions):
+        version_display_name = version.get("display_name", f"v{i}")
+        pipeline_version_name = get_version_name(pipeline_name, version_display_name, i, add_prefix)        
+        pipeline_spec = version.get("pipeline_spec", {})
+        
+        pipeline_version_obj = {
+            "apiVersion": K8S_PIPELINE_API_VERSION,
+            "kind": "PipelineVersion",
+            "metadata": {
+                "name": pipeline_version_name,
+                "namespace": namespace,
+                "annotations": {
+                    "pipelines.kubeflow.org/original-id": original_id,
+                    "pipelines.kubeflow.org/display-name": display_name
```

Comment:
Pipeline versions have their own IDs and display names. Could you use those here instead of the pipeline ones?